Домашняя работа 6

Задание 1.

In [1]:
from math import sqrt 
from scipy.stats import norm
import numpy as np
import pandas as pd

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.

In [3]:
cov = (zp*ks).mean() - zp.mean()*ks.mean()
cov

9157.839999999997

In [4]:
np.cov(zp,ks, ddof =0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [5]:
cov/(zp.std()*ks.std())

0.8874900920739158

In [6]:
df = pd.DataFrame({'zp':zp, 'ks':ks}, columns = [zp, ks])
df

Empty DataFrame
Columns: [(35, 401), (45, 574), (190, 874), (200, 919), (40, 459), (70, 739), (54, 653), (150, 902), (120, 746), (110, 832)]
Index: []

In [7]:
df.cov()

Empty DataFrame
Columns: []
Index: []

Оно ничего не вернуло, я не знаю почему. Смотрел разбор, там вы то же самое делаете. Но ответ при ручном решении совпал. 0.887

Задание 2.

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [8]:
arr = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [9]:
avg = arr.mean()
avg

118.1

Критерий стьюдента для данной задачи - 2.262

In [10]:
std = sqrt(arr.var(ddof =1))
std

10.54566788359614

In [11]:
T1 = avg + (2.262*std)/sqrt(10)
T1

125.64339223691834

In [12]:
T2 = avg - (2.262*std)/sqrt(10)
T2

110.55660776308164

Доверительный интервал - [110.56, 125.64]

Задание 3.

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [13]:
A = 0.95 + (1 - 0.95)/2 
A # Вот здесь у меня есть определенные проблемы. Я не всегда понимаю, надо ли брать просто альфу, или альфу пополам. Если в задаче это не указано, то не понятно отчего отталкиваться. 

0.975

In [14]:
norm.ppf(A)

1.959963984540054

In [15]:
avg = 174.2
D = 25

In [16]:
Z1 = avg + (norm.ppf(A)*sqrt(D))/sqrt(27)
Z1

176.08597622346028

In [17]:
Z2 = avg - (norm.ppf(A)*sqrt(D))/sqrt(27)
Z2

172.3140237765397

Доверительный интервал - [172.31, 176.08]

Курсач делать скорее всего не буду, времени нет. 